blockchain/types/tx_internal_data_ethereum_blob.go

```diff
@@ -221,6 +221,42 @@ func (sc *BlobTxSidecar) Copy() *BlobTxSidecar {
 	}
 }
 
+// ValidateWithBlobHashes validates the sidecar with the blob hashes.
+// This method also verifies the validity of the proof.
+// Validation rules:
+// 1. The BlobTxWithBlobs format complies with EIP-7594.
+// - assert sidecar_version == 1  # BlobSidecarVersionV1
+// - assert len(tx.blobVersionedHashes) == len(blobs) == len(commitments)  # 1 commitment per blob
+// - assert len(blobs) * CELLS_PER_EXT_BLOB == len(proofs)  # 128 proofs per blob
+// 2. The BlobTxWithBlobs contains the correct proofs
+// - for i in range(len(tx.blobVersionedHashes)):
+//   - assert CalcBlobHashV1(commitments[i]) == tx.blobVersionedHashes[i]
+//
+// - assert VerifyCellProofs(blobs, commitments, cellProofs)
+// ref: https://github.com/kaiachain/kips/blob/fac337aad17db40ed2a6c988e03ad6b2ec9771f2/KIPs/kip-279.md#blobtxwithblobs-validation
+func (sc *BlobTxSidecar) ValidateWithBlobHashes(hashes []common.Hash) error {
```

Comment:
Can we cache (part of) the validation? CNs likely to already validated it once when they received blobTxWithBlobs over txpool p2p.
We can uniquely identify a sidecar with its versionedHash.

Reply:
I think we can. Like this?

```
type BlobTxSidecar struct {
	Version     byte                 // Version
	...

	validatedHash common.Hash
}

func (sc *BlobTxSidecar) ValidateWithBlobHashes(hashes []common.Hash) error {
	if hash(hashes) == sc.validatedHash {
		return nil
	}
	...
```